src/solver/batchnorm/backward_ck.cpp

```diff
@@ -161,89 +198,307 @@ template <typename XDataType,
-                                                   MeanVarDataType>,
-                                 CKArgsBNormBwd,
-                                 miopen::batchnorm::BwdInvokeParams>(bn_problem, kernel_id);
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = problem_desc;
+    return false;
+#else
+    if(this->valid_kernels.empty())
+    {
+        this->HeuristicInit(problem_desc);
+        if(valid_kernels.empty())
+            MIOPEN_THROW(miopenStatusInternalError, "BnCKBwdBackward valid_kernels empty");
+        return true;
+    }
+    if((this->index + 1) < valid_kernels.size())
+    {
+        ++this->index;
+        this->kernel_id = this->valid_kernels[index];
+        return true;
+    }
+    else
+        return false;
+#endif
+}
+
+bool PerformanceConfigBnCKBwdBackward::IsValidValue() const
+{
+    return this->index >= 0 && this->index < valid_kernels.size();
 }
 
+bool PerformanceConfigBnCKBwdBackward::IsValid(
+    const ExecutionContext&, const miopen::batchnorm::ProblemDescription& problem_desc) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = problem_desc;
+    return false;
+#else
+    switch(problem_desc.GetXDesc().GetType())
+    {
+    case miopenHalf: return CheckIsSupportCKArgs<F16, F32, F32, F32, F16, F32, F32>(problem_desc);
+    case miopenBFloat16:
+        return CheckIsSupportCKArgs<BF16, F32, F32, F32, BF16, F32, F32>(problem_desc);
+    case miopenFloat: return CheckIsSupportCKArgs<F32, F32, F32, F32, F32, F32, F32>(problem_desc);
+    case miopenDouble: return CheckIsSupportCKArgs<F64, F64, F64, F64, F64, F64, F64>(problem_desc);
+    case miopenFloat8:
+    case miopenBFloat8:
+    case miopenInt8:
+    case miopenInt32:
+    case miopenInt64:
+    default: MIOPEN_THROW("Unsupported datatype");
+    }
+    return false;
 #endif
+}
+
+bool PerformanceConfigBnCKBwdBackward::operator==(
+    const PerformanceConfigBnCKBwdBackward& other) const
+{
+    return this->kernel_id == other.kernel_id;
+}
+
+template <typename XDataType,
+          typename DxDataType,
+          typename DyDataType,
+          typename AccDataType,
+          typename ScaleDataType,
+          typename DscaleDbiasDataType,
+          typename MeanVarDataType>
+static int CheckCKApplicability(const miopen::batchnorm::ProblemDescription& problem)
+{
+    const auto& args       = CKArgsBNormBwd{problem};
+    const auto bn_bwd_ptrs = DeviceOpBNBwdPtrs<XDataType,
+                                               DxDataType,
+                                               DyDataType,
+                                               AccDataType,
+                                               ScaleDataType,
+                                               DscaleDbiasDataType,
+                                               MeanVarDataType>::GetInstances();
+    if(bn_bwd_ptrs.empty())
+        MIOPEN_THROW(miopenStatusInternalError, "BnCKBwdBackward kernel empty");
```

Comment:
It throws in `isApplicable`, is it intended?
I mean is it really serious internal error (like bn_bwd_ptrs must never have been empty), or it just mean that CK is not applicable for some reason?